### PR TITLE
[AppConfig] adding samples based on .NET samples

### PR DESCRIPTION
**Pull request by seankane-msft**
Based on [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample8_ChangeNotification.md) sample.

### CODE REVIEW

**Review by pakrym — APPROVED** [review]

File: sdk/appconfiguration/azure-appconfiguration/samples/sync_token_async_samples.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+async def handle_event_grid_notifications(event_grid_events):
+    # type: (List[dict[str, Any]]) -> None
+    CONNECTION_STRING = get_connection_string()
+
+    all_keys = []
+
+    async with AzureAppConfigurationClient.from_connection_string(CONNECTION_STRING) as client:
+
+        async for event_grid_event in event_grid_events:
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/samples/sync_token_async_samples.py:29]: If event_grid_events is list, we only need "for" here.

File: sdk/appconfiguration/azure-appconfiguration/samples/sync_token_samples.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    all_keys = []
+
+    with AzureAppConfigurationClient.from_connection_string(CONNECTION_STRING) as client:
+
+        for event_grid_event in event_grid_events:
+            if event_grid_event["eventType"] == 'Microsoft.KeyValueModified':
+                sync_token = event['data']['syncToken']
+                client.update_sync_token(sync_token)
+
+                new_key = client.get_configuration_setting(key=event['data']['key'], label=event['data']['label'])
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/samples/sync_token_samples.py:33]: Do we want to support get_config_setting(key, label, sync_token)?

Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/samples/sync_token_samples.py:33]: We can start the discussion with the v-team. Of course we don't want to add it in this release. :)

**Review by xiangyan99 — APPROVED** [review]

**Comment by xiangyan99** [comment]
You may also want to update samples/README.md to add description of your sample

**Pull request merged**